runtime/compiler/runtime/MetricsServer.cpp

```diff
@@ -399,20 +637,21 @@ MetricsServer::reArmSocketForReading(int sockIndex)
    _pfd[sockIndex].revents = 0;
    }
 
+void
+MetricsServer::reArmSocketForWriting(int sockIndex)
+   {
+   _pfd[sockIndex].events = POLLOUT;
+   _pfd[sockIndex].revents = 0;
+   }
+
 void
 MetricsServer::closeSocket(int sockIndex)
    {
    close(_pfd[sockIndex].fd);
    _pfd[sockIndex].fd = -1;
    _pfd[sockIndex].revents = 0;
    _numActiveSockets--;
-
-   if (_incompleteRequests[sockIndex])
-      {
-      // Delete the dynamically allocated data
-      TR_Memory::jitPersistentFree(_incompleteRequests[sockIndex]);
-      _incompleteRequests[sockIndex] = NULL;
-      }
+   _requests[sockIndex].clear();
```

Comment:
OBIO_free_all needs to be done before closing the underlying socket. See https://github.com/eclipse-openj9/openj9/pull/15392

Reply:
It's better to just move ` _requests[sockIndex].clear();` at the top of this function. As it stands, we set `_pfd[sockIndex].fd = -1;` and then try to close descriptor -1.